### PR TITLE
Do not start a CCD event if envelope has no new document

### PR DESCRIPTION
**Pull request by rkondratowicz**
https://tools.hmcts.net/jira/browse/BPS-543

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/AttachDocsToSupplementaryEvidence.java [hunk 1]
```diff
@@ -1,22 +1,37 @@
...
     AttachDocsToSupplementaryEvidence(SupplementaryEvidenceMapper mapper) {
         this.mapper = mapper;
     }
 
+    public void handle(Envelope envelope, CaseDetails existingCase) {
+        log.info("Adding new evidence to case {} from envelope {}", existingCase.getId(), envelope.id);
+
+        if (mapper.getDocsToAdd(getDocuments(existingCase), envelope.documents).isEmpty()) {
+            // an alert relies on this message, do not modify it.
+            log.warn("Envelope has no new documents. CCD Case not updated");
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/AttachDocsToSupplementaryEvidence.java]: I don't recall having such alert

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/AttachDocsToSupplementaryEvidence.java [hunk 2]
```diff
@@ -1,22 +1,37 @@
...
+    private static final Logger log = LoggerFactory.getLogger(AbstractEventPublisher.class);
 
     private final SupplementaryEvidenceMapper mapper;
 
     AttachDocsToSupplementaryEvidence(SupplementaryEvidenceMapper mapper) {
         this.mapper = mapper;
     }
 
+    public void handle(Envelope envelope, CaseDetails existingCase) {
+        log.info("Adding new evidence to case {} from envelope {}", existingCase.getId(), envelope.id);
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/AttachDocsToSupplementaryEvidence.java]: The log of start of event publishing is already present in abstract publisher. Do we want to add extra log here? Maybe we can take extra review on this and record this particular log when we actually begin adding new evidence.

The following if statement is more of an extra validation so we should be doing all of that and once the horizon is clear - record the beginning

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/AttachDocsToSupplementaryEvidence.java [hunk 3]
```diff
@@ -1,22 +1,37 @@
...
 import uk.gov.hmcts.reform.bulkscan.orchestrator.model.ccd.mappers.SupplementaryEvidenceMapper;
 import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.model.Envelope;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
 import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
 
 import static uk.gov.hmcts.reform.bulkscan.orchestrator.helper.ScannedDocumentsHelper.getDocuments;
 
 @Component
 class AttachDocsToSupplementaryEvidence extends AbstractEventPublisher {
+    private static final Logger log = LoggerFactory.getLogger(AbstractEventPublisher.class);
```
[the remark this answers is not in the document]
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/AttachDocsToSupplementaryEvidence.java]: copy paste indeed. fixing.

**Comment by rkondratowicz** [comment]
closing to start with a fresh pipeline... :/

**Pull request closed without merging**